src/main/java/de/retest/recheck/RecheckOptions.java

```diff
@@ -7,74 +7,160 @@
 import de.retest.recheck.ignore.CompoundFilter;
 import de.retest.recheck.ignore.Filter;
 import de.retest.recheck.ignore.RecheckIgnoreUtil;
+import de.retest.recheck.persistence.FileNamer;
 import lombok.AccessLevel;
 import lombok.AllArgsConstructor;
-import lombok.Getter;
 
+/**
+ * This class configures the behaviour of {@link Recheck} and their implementations.
+ */
 @AllArgsConstructor( access = AccessLevel.PRIVATE )
-@Getter
 public class RecheckOptions {
 
 	private final FileNamerStrategy fileNamerStrategy;
 	private final String suiteName;
 	private final boolean reportUploadEnabled;
 	private final Filter filter;
+	private final List<Filter> filterToAdd;
```

Comment:
Could you ELI5 why you moved this stuff into here? I don't quite get the commit message. Thx.

Reply:
I just fear, that the creation of the options and thus the call to `FileNamerStrategy#getTestClass()` might happen too early (i. e. outside the scope of a test) and thus might result in a wrong name or suite location (ignore). We cannot know, how the `FileNamerStrategy` is implemented.

Thus making sure that this code is executed in the getters ensures we are in the context of `Recheck`.

Reply:
Personally, I like that `RecheckOptions` is just a data object without logic. The "complicated" part is creating the options, which is why I prefer this stuff to be inside the builder.

However, I get your point and valid arguments. I'm just trying to think of a situtation where options might be created where a test class is _not_ part of the stack trace. If someone doesn't use JUnit or TestNG, they have to provide a `FileNamerStrategy` anyway?

Reply:
I talked to @roesslerj&mdash;as you both reworked the filter&mdash;and he approves this change, to (somehow) lazily get the correct values within `RecheckOptions`.

I agree with you that the builder should be the logic part. However, the main part is the `suiteName` which couples to the `FileNamerStrategy` and loads the filter. Thus, I think, it is more important to ensure the correct context to be sure about this name. If we can ensure that the builder is always in this context, that's fine. Maybe its just my paranoia running wild ... 😉.

Reply:
Allright. But may I suggest to keep the single `Filter` field in the options class? This is really confusing and can be safely set via the builder.

Reply:
That's the best, I can do... We need some kind of indicator, since we have two possibilities. Either the user overwrites the filter (-> do not add the filters) or the user wants to append only filters (-> add the filters). In both cases, the builder would generate a single `Filter`.

But the default filters need the `suiteName` which was the reason, I re-did this in the first place.